### DISCUSSION TITLE
pretraind model을 사용할 수 있는 경우?

### DISCUSSION THREAD

**Discussion by ddobokki** [Boostcamp-AI-Tech-1-15/ddobokki_log#14] [3. Pytorch]
개와 고양이를 구분하는 모델을 만든다고 가정하고 누군가 빅 데이터를 활용해 개와 고양이가 포함된 100가지의 동물을 구분하는 모델을 만들었다고 하면 그 모델에 레이어를 추가하거나 최종 레이어를 수정 하는 것으로 tranfer learning을 할 수 있습니다.

하지만 만약 많은 데이터로 학습을 시켰는데 개와 고양이가 포함되지 않는 데이터일 경우(가령 mnist)의 경우에도 tranfer learning이 효과를 볼 수 있을까요? 아니면 완전히 random initialize 한것과 큰 차이가 없을까요?

**Comment by minji-o-j** [comment]
후자(random initialize) 한 것 아닐까요?? 네트워크 자체가 input class에 맞춰지니까요!!

찾아보니까 새로운 class이면 처음부터 훈련시키는 게 맞는것같아요!
https://www.quora.com/Can-we-use-a-pre-trained-model-on-a-dataset-with-different-classes-than-the-ones-it-was-trained-on

그리고 class가 같더라도, 원본 data와 data 분포가 비슷할 때 더 잘 적용된다고 합니다

그리고 원본 class에 class를 추가하는 경우에도 [이렇다](https://stackoverflow.com/a/60019168) 고 하는데 저도 class추가하는 것만 해봤을 때는 데이터를 처음부터 훈련시키는 것보다 pretrained model에 finetuning시키는게 더 성능이 좋았던 것으로 기억합니다..!!

**Comment by mybirth0407** [comment]
Case by case인데요. 기본적으로 어떤 이미지, 어떤 물체, 이미지의 어떤 특징을 예측하는 방향으로 학습이 되기 때문에 기본적으로 pretrained 모델이 성능이 좋습니다. feature extraction 개념에서 이미 어느정도 완성이 되어 있기 때문에 그렇습니다.  하지만 클래스가 너무 커지는 경우 or data distribution이 완전히 다른 data가 메인일 경우 오히려 성능이 떨어질 때가 있는데요. 제 경험으로는 imagenet pretrained 모델에서 자동차 관련 task를 진행했을 때 pretrained 모델이 성능이 더 떨어지는 경우가 있었습니다.
